From duncan.sands@free.fr Wed Sep 27 14:39:33 2006
From: Duncan Sands <duncan.sands@free.fr>
To: Greg KH <greg@kroah.com>
Subject: Driver core: plug device probe memory leak
Date: Wed, 27 Sep 2006 23:38:08 +0200
Cc: Cedric Le Goater <clg@fr.ibm.com>
Content-Disposition: inline
Message-Id: <200609272338.09416.duncan.sands@free.fr>

Make sure data is freed if the kthread fails to start.

Signed-off-by: Duncan Sands <baldrick@free.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/dd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/base/dd.c
+++ gregkh-2.6/drivers/base/dd.c
@@ -178,7 +178,7 @@ int driver_probe_device(struct device_dr
 		probe_task = kthread_run(really_probe, data,
 					 "probe-%s", dev->bus_id);
 		if (IS_ERR(probe_task))
-			ret = PTR_ERR(probe_task);
+			ret = really_probe(data);
 	} else
 		ret = really_probe(data);
 
